player/lua/ytdl_hook.lua

```diff
@@ -754,6 +755,36 @@ local function add_single_video(json)
         end
     end
 
+    -- add thumbnails
+    if (o.thumbnails == 'all' or o.thumbnails == 'best') and not (json.thumbnails == nil) then
+        local thumb = nil
+        local thumb_height = -1
+        local thumb_preference = nil
+
+        for i = #json.thumbnails, 1, -1 do
+            local thumb_info = json.thumbnails[i]
+            if not (thumb_info.url == nil) then
+                if (o.thumbnails == 'all') then
+                    msg.verbose("adding thumbnail")
+                    mp.commandv("video-add", thumb_info.url, "auto")
+                    thumb_height = 0
+                elseif (thumb_preference ~= nil and thumb_info.preference > thumb_preference) or
+                    (thumb_preference == nil and ((thumb_info.height or 0) > thumb_height)) then
+                    thumb = thumb_info.url
+                    thumb_height = thumb_info.height or 0
+                    thumb_preference = thumb_info.preference
+                end
+            end
+        end
+
+        if not (thumb == nil) then
```

Comment:
I'm not lua expert, so I'm mostly asking for myself, why do `if not (thumb == nil)` instead `if thumb ~= nil` or even `not thumb` to filter out if it expand to `false`?

Reply:
Because that's what the code around it does. I don't like it either.
Also that specific one wasn't written by me, so I'm not going to change it unless there is a need to.

Reply:
Yeah this entire file has always been in this style AFAIK. Could be changed later if someone really cares.

Reply:
I see, thanks.